msgraph/core/middleware/telemetry.py

```diff
@@ -0,0 +1,88 @@
+import platform
+
+from urllib3.util import parse_url
+
+from .._constants import SDK_VERSION
+from .._enums import NationalClouds
+from .middleware import BaseMiddleware
+
+ENDPOINTS = {
```

Comment:
Since `NationalClouds` is a string `Enum` we can't directly check for membership ising the `in` operand. Hence the need to use the actual string values corresponding to the endpoints. I have however updated the logic to be more efficient.

Reply:
Since `NationalClouds` is a string `Enum` we can't directly check for membership using the `in` operand. Hence the need to use the actual string values corresponding to the endpoints. I have however updated the logic to be more efficient.

Reply:
Since `NationalClouds` is a string `Enum` we can't directly check for membership using the `in` operand. Hence the need to use the actual string values corresponding to the endpoints. I have however updated the logic to be more efficient.